src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -155,67 +163,224 @@ private void InitializeDefaultAutoCompleteCandidates()
+                    {
+                        nodeRequest.Type.Id = node.CreationName;
+                    }
+                    else if (nodeToAdd is NodeModel nodeModel)
+                    {
+                        string typeId = GenerateTypeIdForNodeModelNode(nodeModel.GetType().AssemblyQualifiedName);
+                        nodeRequest.Type.Id = typeId;
+                    }
+                    request.Context.Nodes = request.Context.Nodes.Append(nodeRequest);
+                }
             }
-            FilteredResults.ToList()[0].Model.AutoCompletionNodeMachineLearningInfo.ConfidenceScore = 30;
-            FilteredResults.ToList()[1].Model.AutoCompletionNodeMachineLearningInfo.ConfidenceScore = 40;
 
-            if (!FilteredResults.Where(n => n.Model.AutoCompletionNodeMachineLearningInfo.ConfidenceScore >= 50).Any())
+            string jsonRequest = JsonConvert.SerializeObject(request);
+            MLNodeAutoCompletionResponse MLresults = null;
+
+            // Get results from the ML API.
+            try
             {
-                FilteredResults = new List<NodeSearchElementViewModel>();
+                MLresults = GetMLNodeAutocompleteResults(jsonRequest);
+            }
+            catch(Exception) {
                 DisplayNoRecommendationsLowConfidence = true;
-                NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteLowConfidenceTitle;
-                NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteLowConfidenceMessage;
-                LowConfidenceMessageAdditional = Resources.AutocompleteLowConfidenceMessageAditional;
-                DisplayLowConfidence = true;
+                DisplayLowConfidence = false;
+                NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
+                NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
+                return;
             }
-            */
+
+            var matchingResults = new List<NodeSearchElement>();
+            FilteredResults = new List<NodeSearchElementViewModel>();
+            var zeroTouchSearchElements = Model.SearchEntries.OfType<ZeroTouchSearchElement>().Where(x => x.IsVisibleInSearch);
+            var nodeModelSearchElements = Model.SearchEntries.OfType<NodeModelSearchElement>().Where(x => x.IsVisibleInSearch);
+
+            // no results
+            if (MLresults == null || MLresults.Results.Count() == 0)
+            {
+                DisplayNoRecommendationsLowConfidence = true;
+                DisplayLowConfidence = false;
+                NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
+                NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
+                return;
+            }
+
+            // ML Results are categorized based on the threshold confidence score before displaying. 
```

Comment:
Currently the service returns no nodes coming from packages, but we want to enable package support in the next month. If the recommendation list contains a node from a package (that the user has loaded), would the search for the `NodeSearchElementViewModel` still work?

Reply:
yes as long as the node info is found in the VM, it should work.